src/plantuml.cpp

```diff
@@ -95,34 +73,153 @@ QCString PlantumlManager::writePlantUMLSource(const QCString &outDirArg,const QC
             text+=' ';
             text+=imgName;
+
+            if (!rest.isEmpty())
+            {
+              text += '\n';
+              text += rest;
+            }
             if (c) text+=c;
           }
+          else if (!insideComment && strncmp(p,("end"+locEngine).data(), 3+strlen(engine.data()))==0) // @end...
+          {
+            text += "end"+locEngine+"\n";
+            p+=3+locEngine.length();
+            if (!inlineCode)
+            {
+              QCString qcOutDir(substitute(outDir,"\\","/"));
+              uint32_t pos = qcOutDir.findRev("/");
+              QCString generateType(qcOutDir.right(qcOutDir.length() - (pos + 1)) );
+              Debug::print(Debug::Plantuml,0,"*** writePlantUMLSource generateType: {}\n",generateType);
+              PlantumlManager::instance().insert(generateType.str(),puName.str(),outDir,format,text,srcFile,srcLine);
+              Debug::print(Debug::Plantuml,0,"*** writePlantUMLSource generateType: {}\n",generateType);
+              baseNameVector.emplace_back(baseName);
+              text.clear();
+            }
+          }
           break;
         default:
-          if (!insideComment) initial=false;
           break;
       }
     }
     text+='\n';
   }
-  if (inlineCode) text +="@end"+engine+"\n";
+  if (inlineCode)
+  {
+    text +="@end"+engine+"\n";
+    //printf("content\n====\n%s\n=====\n->\n-----\n%s\n------\n",qPrint(content),qPrint(text));
+    QCString qcOutDir(substitute(outDir,"\\","/"));
+    uint32_t pos = qcOutDir.findRev("/");
+    QCString generateType(qcOutDir.right(qcOutDir.length() - (pos + 1)) );
+    Debug::print(Debug::Plantuml,0,"*** writePlantUMLSource generateType: {}\n",generateType);
+    PlantumlManager::instance().insert(generateType.str(),puName.str(),outDir,format,text,srcFile,srcLine);
+    Debug::print(Debug::Plantuml,0,"*** writePlantUMLSource generateType: {}\n",generateType);
+    baseNameVector.emplace_back(baseName);
+  }
 
-  //printf("content\n====\n%s\n=====\n->\n-----\n%s\n------\n",qPrint(content),qPrint(text));
+  return baseNameVector;
+}
 
-  QCString qcOutDir(substitute(outDir,"\\","/"));
-  uint32_t pos = qcOutDir.findRev("/");
-  QCString generateType(qcOutDir.right(qcOutDir.length() - (pos + 1)) );
-  Debug::print(Debug::Plantuml,0,"*** writePlantUMLSource generateType: {}\n",generateType);
-  PlantumlManager::instance().insert(generateType.str(),puName.str(),outDir,format,text,srcFile,srcLine);
-  Debug::print(Debug::Plantuml,0,"*** writePlantUMLSource generateType: {}\n",generateType);
+void PlantumlManager::generatePlantUmlFileNames(const QCString &fileName,OutputFormat format,const QCString &outDir,
+                                                QCString &baseName,QCString &puName,QCString &imgName)
+{
+  static int umlindex=1;
+
+  if (fileName.isEmpty()) // generate name
+  {
+    std::lock_guard<std::mutex> lock(g_PlantUmlMutex);
+    puName = "inline_umlgraph_"+QCString().setNum(umlindex);
```

Comment:
Using numbers like this will give instable output, i.e. when using multiple threads every run the numbers can be assigned to different graphs, and thus the output will change even if the input did not. A typical solution is to use a hash of the filename + line number, representing where the graph is found in the input, and use that to create the output name.

Reply:
The original was already using the non stable numbers and this could lead to multiple use of the same number (it is unlikely that this happens, but better to be safe hence the mutex). It indeed would be better to use a hash.